lambdas/src/core/resource/pipelines/endpoint.rs

```diff
@@ -0,0 +1,121 @@
+use super::adaptor::{
+    get_pipelines_bna_adaptor, get_pipelines_bnas_adaptor, patch_pipelines_bna_adaptor,
+    post_pipelines_bna_adaptor,
+};
+use super::schema::{BnaPipeline, BnaPipelinePatch, BnaPipelinePost, BnaPipelines};
+use crate::{core::resource::schema::ErrorResponses, Context, ExecutionError};
+use axum::{
+    extract::{Path, Query},
+    http::StatusCode,
+    Json,
+};
+use effortless::api::PaginationParameters;
+use entity::wrappers::bna_pipeline::{BNAPipelinePatch, BNAPipelinePost};
+use serde_json::{json, Value};
+use tracing::debug;
+use utoipa_axum::{router::OpenApiRouter, routes};
+use uuid::Uuid;
+
+const TAG: &str = "pipeline";
+
+pub fn routes() -> OpenApiRouter {
+    OpenApiRouter::new()
+        .routes(routes!(get_pipelines_bna))
+        .routes(routes!(get_pipelines_bnas))
+        .routes(routes!(post_pipelines_bna))
+        .routes(routes!(patch_pipelines_bna))
+}
+
+#[utoipa::path(
+  get,
+  path = "/pipelines/bna/{pipeline_id}",
+  description = "Get the details of a specific BNA pipeline",
+  tag = TAG,
+  params(
+    ("pipeline_id" = Uuid, Path, description = "Pipeline identifier")
+  ),
+  responses(
+    (status = OK, description = "Fetches the details of a BNA pipeline", body = BnaPipeline),
+    ErrorResponses,
+  ))]
+async fn get_pipelines_bna(
+    Path(pipeline_id): Path<Uuid>,
+    ctx: Context,
+) -> Result<Json<BnaPipeline>, ExecutionError> {
+    get_pipelines_bna_adaptor(pipeline_id, ctx)
+        .await
+        .map(BnaPipeline::from)
+        .map(Json)
+}
+
+#[utoipa::path(
+  get,
+  path = "/pipelines/bna",
+  description = "Get the details of a specific BNA pipeline",
```

Comment:
i think the description needs to be updated? this is a fetch all request?

Reply:
Yep! Good catch again!